seed/seedtest/seedtest.go

```diff
@@ -0,0 +1,176 @@
+		"series":       "16",
+		"snap-id":      snapID,
+		"publisher-id": developerID,
+		"snap-name":    snapName,
+		"timestamp":    time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	sha3_384, size, err := asserts.SnapFileSHA3_384(snapFile)
+	c.Assert(err, IsNil)
+
+	revA, err := ss.StoreSigning.Sign(asserts.SnapRevisionType, map[string]interface{}{
+		"snap-sha3-384": sha3_384,
+		"snap-size":     fmt.Sprintf("%d", size),
+		"snap-id":       snapID,
+		"developer-id":  developerID,
+		"snap-revision": revision.String(),
+		"timestamp":     time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	if !revision.Unset() {
+		info.SnapID = snapID
+		info.Revision = revision
+	}
+
+	if ss.DB != nil {
+		err := ss.DB.Add(declA)
+		c.Assert(err, IsNil)
+		err = ss.DB.Add(revA)
+		c.Assert(err, IsNil)
+	}
+
+	if ss.snaps == nil {
+		ss.snaps = make(map[string]string)
+		ss.infos = make(map[string]*snap.Info)
+	}
+
+	ss.snaps[snapName] = snapFile
+	ss.infos[snapName] = info
+
+	return declA.(*asserts.SnapDeclaration), revA.(*asserts.SnapRevision)
+}
+
+func (ss *SeedSnaps) AssertedSnap(snapName string) (snapFile string) {
+	return ss.snaps[snapName]
+}
+
+func (ss *SeedSnaps) AssertedSnapInfo(snapName string) *snap.Info {
+	return ss.infos[snapName]
+}
+
+// TestingSeed helps setting up a populated testing seed.
+type TestingSeed struct {
+	SeedSnaps
+
+	SnapsDir   string
+	AssertsDir string
+}
+
+func (s *TestingSeed) MakeAssertedSnap(c *C, snapYaml string, files [][]string, revision snap.Revision, developerID string) (snapFname string, snapDecl *asserts.SnapDeclaration, snapRev *asserts.SnapRevision) {
+	decl, rev := s.SeedSnaps.MakeAssertedSnap(c, snapYaml, files, revision, developerID)
+
+	snapFile := s.snaps[decl.SnapName()]
+
+	snapFname = filepath.Base(snapFile)
+	targetFile := filepath.Join(s.SnapsDir, snapFname)
+	err := os.Rename(snapFile, targetFile)
+	c.Assert(err, IsNil)
+
+	return snapFname, decl, rev
+}
+
+func (s *TestingSeed) MakeModelAssertionChain(brandID, model string, extras ...map[string]interface{}) []asserts.Assertion {
+	assertChain := []asserts.Assertion{}
+	modelA := s.Brands.Model(brandID, model, extras...)
+
+	assertChain = append(assertChain, s.Brands.Account(modelA.BrandID()))
+	assertChain = append(assertChain, s.Brands.AccountKey(modelA.BrandID()))
+	assertChain = append(assertChain, modelA)
+
+	storeAccountKey := s.StoreSigning.StoreAccountKey("")
+	assertChain = append(assertChain, storeAccountKey)
+	return assertChain
+}
+
+func (s *TestingSeed) WriteAssertions(fn string, assertions ...asserts.Assertion) {
+	multifn := filepath.Join(s.AssertsDir, fn)
+	f, err := os.OpenFile(multifn, os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+	enc := asserts.NewEncoder(f)
+	for _, a := range assertions {
+		err := enc.Encode(a)
+		if err != nil {
+			panic(err)
```

Comment:
I'd slightly prefer `c.Assert` instead of panic here and above (and make WriteAssertion take `c *C`), but not biggie (and I realize this code was just moved around).